### PR TITLE
docs: Add dev docs

### LINKED ISSUE #206: Add developer docs
Hi,

I managed to get the project setup locally, but it was not straightforward because the docs cover end-users rather than developers.

Can I suggest `docs/development.md`.

### PR DESCRIPTION
**Pull request by MichaelCurrin**
Fixes #206

### CODE REVIEW

File: docs/development.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+### Open interactive console
+
+```bash
+$ script/console
+```
+
+### Test
+
+Run all unit tests:
```
Comment (lardieri) [inline docs/development.md:71]: For consistent punctuation, this should be

`Run all unit tests.`

File: docs/development.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+```
+
+Run specific unit tests.
+
+```bash
+$ script/test PATH
+$ # e.g.
+$ script/test spec/owner_spec.rb
+```
+
```
Comment (lardieri) [inline docs/development.md:84]: May I suggest the following patterns as well? I found these useful when I started contributing to this project and had to diagnose CI/CD failures and add new test cases. But for someone like myself who does not use Ruby or Rspec professionally, they are not obvious.

Run tests in the order they are written (not random).

```bash
$ script/test --order defined
```

Run tests in the same random order as a previous run.
```bash
$ script/test --seed 12345
```

Run tests with verbose trace logs.
```bash
$ script/test --format documentation
```

File: docs/development.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+$ script/bootstrap
+```
+
+## Usage
+
+See the [script](/script/) directory.
+
+### Format
+
+Check formatting.
```
Comment (lardieri) [inline docs/development.md:51]: May I suggest giving more of a hint to new contributors? For example:

```
Check code formatting before committing.
```

**Review by lardieri — CHANGES_REQUESTED** [review]

File: docs/development.md (hunk 1 above)
Comment (MichaelCurrin) [inline docs/development.md:71]:
```suggestion
Run all unit tests.
```

File: docs/development.md (hunk 3 above)
Comment (MichaelCurrin) [inline docs/development.md:51]:
```suggestion
Check for code formatting issues - recommended before you commit.
```

File: docs/development.md (hunk 2 above)
Comment (MichaelCurrin) [inline docs/development.md:84]:
```suggestion

Run tests in the order they are written (not a random order).

```bash
$ script/test --order defined
```

Run tests in the same random order as a previous run. Here with a sample seed value.

```bash
$ script/test --seed 12345
```

Run tests with verbose trace logs.

```bash
$ script/test --format documentation
```

```

Comment (MichaelCurrin) [reply docs/development.md:82]: Thanks, I've added.

**Comment by MichaelCurrin** [comment]
Thanks for suggestions. I've applied and also fixed up the headings (I'm using to having H1 with H2s below, so had to fix it up to have one H2 and multiple H3s to match other docs).

**Review by lardieri — APPROVED** [review]

**Comment by lardieri** [comment]
Thank you for writing this document. I wish I had had it when I started using GitHub Pages six months ago!

File: docs/development.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+#### Requirements
+
+- Ruby
+- Bundler
+
+#### Install system dependencies
+
+- Install Ruby - see the [Downloads](https://www.ruby-lang.org/en/downloads/) page.
+- Install Bundler - see [Bundler](https://bundler.io/) homepage.
```
Comment (mattr-) [inline docs/development.md:14]:
```suggestion
- Install Bundler - see the [Bundler](https://bundler.io/) homepage.
```

File: docs/development.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
+## Development
+> Guide to local development of this plugin
```
Comment (mattr-) [inline docs/development.md:2]: It's not obvious to me why this is quoted. What do you think about this small formatting change?
```suggestion

Guide to local development of this plugin
```

**Review by mattr- — CHANGES_REQUESTED** [review]
I left two small comments on this PR that I think should be addressed before merging. Thanks for taking a look!

File: docs/development.md (hunk 5 above)
Comment (MichaelCurrin) [reply docs/development.md:2]: It's meant to be a subtitle. It's a pattern commonly followed in README.md files on GitHub. Though not consistently used.

I'll remove.

Comment (MichaelCurrin) [reply docs/development.md:2]: e.g. https://github.com/MichaelCurrin/jekyll-blog-demo#jekyll-blog-demo---

**Review by parkr — APPROVED** [review]

**Comment by parkr** [comment]
@jekyllbot: merge +docs

**Pull request merged**